x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Subject.java

```diff
@@ -47,6 +47,7 @@ public class Subject {
     public enum Type {
         USER,
         API_KEY,
+        CLOUD_API_KEY,
```

Comment:
We have to check and handle BWC implications with addition of the new subject type and realm.
It's a bit awkward that certain implementation details of a custom cloud API key authentication are in the core, but I don't see a simple way (at the moment) that we can avoid this without massive authentication refactoring.

Reply:
Given that this PR only introduces the new subject type without actually using it, I think it should be fine to handle the BWC and introduce a new transport version in a followup PR.